lang/funcs/collection_test.go

```diff
@@ -2213,6 +2236,66 @@ func TestMerge(t *testing.T) {
 			}),
 			false,
 		},
+		{ // merge objects of various shapes
+			[]cty.Value{
+				cty.ObjectVal(map[string]cty.Value{
+					"a": cty.ListVal([]cty.Value{
+						cty.StringVal("b"),
+					}),
+				}),
+				cty.ObjectVal(map[string]cty.Value{
+					"d": cty.DynamicVal,
+				}),
+			},
+			cty.ObjectVal(map[string]cty.Value{
+				"a": cty.ListVal([]cty.Value{
+					cty.StringVal("b"),
+				}),
+				"d": cty.DynamicVal,
+			}),
+			false,
+		},
+		{ // merge maps and objects
+			[]cty.Value{
+				cty.MapVal(map[string]cty.Value{
+					"a": cty.ListVal([]cty.Value{
+						cty.StringVal("b"),
+					}),
+				}),
+				cty.ObjectVal(map[string]cty.Value{
+					"d": cty.NumberIntVal(2),
+				}),
+			},
+			cty.ObjectVal(map[string]cty.Value{
+				"a": cty.ListVal([]cty.Value{
+					cty.StringVal("b"),
+				}),
+				"d": cty.NumberIntVal(2),
+			}),
+			false,
+		},
+		{ // attr a type and value is overridden
```

Comment:
^^ I think we should add this (type _and_ value being overridden) as an example to the documentation, what do you think?